### ISSUE TITLE
[NodeJS18 Upgrade] Update support matrix for node18

### ISSUE THREAD

**Issue by hdhalter** [opensearch-project/documentation-website#4303]
We are going to stay with node16 for 2.9, and plan to move to node18 for 2.10.
This would ensure us to have enough time for the upgrade.
As well as compensate the changes for multiple teams.

We can also start the Python3.9 upgrade process as Python3.7 is about to be eol by 2023/06/23.

Once we go to node18 in 2.10, the support matrix of our product will need to remove these OSes:

* RHEL7&CentOS7 / AmazonLinux2 (glibc2.17 / glibc2.26)
* Ubuntu 16.04 / Ubuntu 18.04 (glibc2.23 / glibc2.27)
(Note: we still bundle node14 as fallback so technically above can still run, but 14 is eol and we do not officially support them anymore)

This also requires that the 2.x branch stays on 16.20.0 in .nvmrc, and have no node18 specific code committed until 2.9.0 is released to the public.

Infra will take care of the node18 upgrade once after 2.9.0.

Thanks.

_Originally posted by @peterzhuamazon in https://github.com/opensearch-project/opensearch-build/issues/1563#issuecomment-1588162950_

**Comment by hdhalter** [comment]
@peterzhuamazon - Is this still on track for 2.10?

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon - Is this still on track for 2.10?

Yes @hdhalter and we will need to change the matrix before 2.10 release.

Thanks.

**Comment by peterzhuamazon** [comment]
New PR:
* https://github.com/opensearch-project/documentation-website/pull/4980

**Comment by natebower** [comment]
Closing as stale
